lib/widgets/autocomplete.dart

```diff
@@ -0,0 +1,174 @@
+        replacementString = '${mention(store.users[userId]!, silent: intent.query.silent)} ';
+      case WildcardMentionAutocompleteResult():
+        replacementString = '[unimplemented]'; // TODO
+      case UserGroupMentionAutocompleteResult():
+        replacementString = '[unimplemented]'; // TODO
+    }
+
+    widget.controller.value = intent.textEditingValue.replaced(
+      TextRange(
+        start: intent.syntaxStart,
+        end: intent.textEditingValue.selection.end),
+      replacementString,
+    );
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final store = PerAccountStoreWidget.of(context);
+    return RawAutocomplete<MentionAutocompleteResult>(
+      textEditingController: widget.controller,
+      focusNode: widget.focusNode,
+      optionsBuilder: (_) => _resultsToDisplay,
+      // RawAutocomplete passes these when it calls optionsViewBuilder:
+      //   AutocompleteOnSelected<T> onSelected,
+      //   Iterable<T> options,
+      //
+      // We ignore them:
+      // - `onSelected` would cause some behavior we don't want,
+      //   such as moving the cursor to the end of the compose-input text.
+      // - `options` would be needed if we were delegating to RawAutocomplete
+      //   the work of creating the list of options. We're not; the
+      //   `optionsBuilder` we pass is just a function that returns
+      //   _resultsToDisplay, which is computed with lots of help from
+      //   MentionAutocompleteView.
+      optionsViewBuilder: (context, _, __) {
+        return Align(
+          alignment: Alignment.bottomLeft,
+          child: Material(
+            elevation: 4.0,
+            child: ConstrainedBox(
+              constraints: const BoxConstraints(maxHeight: 300), // TODO not hard-coded
+              child: ListView.builder(
+                padding: EdgeInsets.zero,
+                shrinkWrap: true,
+                itemCount: _resultsToDisplay.length,
+                itemBuilder: (BuildContext context, int index) {
+                  final option = _resultsToDisplay.elementAt(index);
+                  String? label;
+                  switch (option) {
+                    case UserMentionAutocompleteResult(:var userId):
+                      label = store.users[userId]!.fullName;
+                    case WildcardMentionAutocompleteResult():
+                      label = '[unimplemented]'; // TODO
+                    case UserGroupMentionAutocompleteResult():
+                      label = '[unimplemented]'; // TODO
+                  }
+                  return InkWell(
+                    onTap: () {
+                      _onTapOption(option);
+                    },
+                    child: Builder(
+                      builder: (BuildContext context) {
+                        return Container(
+                          padding: const EdgeInsets.all(16.0),
+                          child: Text(label ?? 'unknown'));
```

Comment:
nit:
```suggestion
                          child: Text(label));
```
and `String label;` above